core/src/test/java/cz/fi/muni/pv217/employee/management/resources/WorkdayResourceTest.java

```diff
@@ -0,0 +1,311 @@
+import cz.fi.muni.pv217.employee.management.providers.MockVacationServer;
+import io.quarkus.test.common.QuarkusTestResource;
+import io.quarkus.test.common.http.TestHTTPEndpoint;
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.response.Response;
+import net.minidev.json.JSONObject;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import javax.transaction.Transactional;
+
+import static cz.fi.muni.pv217.employee.management.util.TestUtil.*;
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.is;
+
+@QuarkusTest
+@QuarkusTestResource(MockKeycloakTestResource.class)
+@QuarkusTestResource(MockVacationServer.class)
+@TestHTTPEndpoint(WorkdayResource.class)
+public class WorkdayResourceTest {
+
+    @BeforeAll
+    @Transactional
+    public static void init(){
+        EMPLOYEE.persist();
+        ORDER.persist();
+        WORKDAY1.persist();
+        WORKDAY2.persist();
+        createAnotherWorkday().persist();
+    }
+
+    @Test
+    @Transactional
+    void testCreateWorkdayWhenEmployeeHasNotVacation() {
+        JSONObject employee = new JSONObject();
+        employee.put("id", "1");
+        employee.put("username", "martin");
+        employee.put("name", "Martin");
+        employee.put("surname", "Hrasko");
+        employee.put("dateOfBirth", "1990-01-01");
+        employee.put("insuranceCompany", "Union");
+        employee.put("mobile", "0900000000");
+        employee.put("address", "Brno");
+        employee.put("startContract", "2020-11-11");
+        employee.put("hourlyWage", "5");
+
+        JSONObject order = new JSONObject();
+        order.put("id", "2");
+        order.put("name", "Jozef");
+        order.put("surname", "Orac");
+        order.put("mobile", "0910000000");
+        order.put("state", "INPROGRESS");
+        order.put("info", "Zasad mrkvu");
+
+        JSONObject requestParams = new JSONObject();
+        requestParams.put("date", "2020-12-28");
+        requestParams.put("hours", "8");
+        requestParams.put("employee", employee);
+        requestParams.put("order", order);
+
+        // Do request and save it
+        Response response =
+                given()
+                        .when().auth().oauth2(
+                        getAccessToken("adam"))
+                        .contentType("application/json")
+                        .body(requestParams.toJSONString())
+                        .post("/create");
+        // Check the response
+        response.then()
+                .statusCode(200)
+                .body(    "date", is("2020-12-28"),
+                        "hours", is(8F),
+                        "employee.id", is(1),
+                        "order.id", is(2));
+        // Delete the created workday because transactions suck in integration tests
+        Long id = Long.valueOf((Integer) response.jsonPath().get("id"));
```

Comment:
I have found the way how to get rid of this in my other pull request !3, it is not required but if you want you can use that method to clean DB

Reply:
Yeah, but my tests had worked when I noticed your PR . I don't want to spend any more time with that.